core/corehttp/gateway_handler.go

```diff
@@ -88,15 +88,28 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 
 	urlPath := r.URL.Path
 
-	if i.config.BlockList != nil && i.config.BlockList.ShouldBlock(urlPath) {
-		w.WriteHeader(http.StatusForbidden)
-		w.Write([]byte("403 - Forbidden"))
+	nd, err := core.Resolve(ctx, i.node, path.Path(urlPath))
+	if err != nil {
+		webError(w, "Path Resolve error", err, http.StatusBadRequest)
 		return
 	}
 
-	nd, err := core.Resolve(ctx, i.node, path.Path(urlPath))
+	k, err := nd.Key()
 	if err != nil {
-		webError(w, "Path Resolve error", err, http.StatusBadRequest)
+		log.Error("failed to get key from node: %s", err)
+		webError(w, "Marshaling Error", err, http.StatusInternalServerError)
+		return
+	}
+
+	if i.config.DenyList != nil && i.config.DenyList.Has(k) {
+		w.WriteHeader(451)
+		w.Write([]byte("451 - Unavailable For Legal Reasons"))
```

Comment:
not always. we might have to block things that are not illegal. probably should depend on the blocked thing... ok to merge, but this is not correct in the long run